x-pack/plugins/viz_editor/common/select_builder.ts

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { DatasourceField } from './datasource';
+import { SelectOperation, SelectOperator } from './query_types';
+
+type OperationTemplate = { [operation in SelectOperator]: SelectOperation };
+
+const operationTemplate: OperationTemplate = {
+  count: { operation: 'count' },
+  terms: { operation: 'terms', argument: { field: '', size: 5 } },
+  avg: { operation: 'avg', argument: { field: '' } },
+  sum: { operation: 'sum', argument: { field: '' } },
+  column: { operation: 'column', argument: { field: '' } },
+  date_histogram: {
+    operation: 'date_histogram',
+    argument: {
+      field: '',
+      interval: 'auto',
```

Comment:
This doesn't work at the moment, because `auto` intervals are currently a thing handled by Kibana itself, but it will with https://github.com/elastic/kibana/pull/34152

Reply:
I was wondering why this didn't work, thanks for the explanation